core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -0,0 +1,371 @@
+        }
+
+        override def receiveWithLogging: Receive = {
+          case AkkaAssociatedEvent(_, remoteAddress, _) =>
+            safelyCall(endpoint) {
+              val event = AssociatedEvent(akkaAddressToRpcAddress(remoteAddress))
+              val pf = endpoint.receive
+              if (pf.isDefinedAt(event)) {
+                pf.apply(event)
+              }
+            }
+
+          case AkkaDisassociatedEvent(_, remoteAddress, _) =>
+            safelyCall(endpoint) {
+              val event = DisassociatedEvent(akkaAddressToRpcAddress(remoteAddress))
+              val pf = endpoint.receive
+              if (pf.isDefinedAt(event)) {
+                pf.apply(event)
+              }
+            }
+
+          case AssociationErrorEvent(cause, localAddress, remoteAddress, inbound, _) =>
+            safelyCall(endpoint) {
+              val event = NetworkErrorEvent(akkaAddressToRpcAddress(remoteAddress), cause)
+              val pf = endpoint.receive
+              if (pf.isDefinedAt(event)) {
+                pf.apply(event)
+              }
+            }
+          case e: RemotingLifecycleEvent =>
+          // TODO ignore?
+
+          case AkkaMessage(message: Any, reply: Boolean)=>
+            logDebug("Received RPC message: " + AkkaMessage(message, reply))
+            safelyCall(endpoint) {
+              val s = sender()
+              val pf =
+                if (reply) {
+                  endpoint.receiveAndReply(new RpcResponse {
+                    override def fail(e: Throwable): Unit = {
+                      s ! AkkaFailure(e)
+                    }
+
+                    override def reply(response: Any): Unit = {
+                      s ! AkkaMessage(response, false)
+                    }
+
+                    override def replyWithSender(response: Any, sender: RpcEndpointRef): Unit = {
+                      s.!(AkkaMessage(response, true))(
+                        sender.asInstanceOf[AkkaRpcEndpointRef].actorRef)
+                    }
+                  })
+                } else {
+                  endpoint.receive
+                }
+              try {
+                if (pf.isDefinedAt(message)) {
+                  pf.apply(message)
+                }
+              } catch {
+                case NonFatal(e) =>
+                  if (reply) {
+                    // If the sender asks a reply, we should send the error back to the sender
+                    s ! AkkaFailure(e)
+                  } else {
+                    throw e
+                  }
+              }
+            }
+          case message: Any => {
+            logWarning(s"Unknown message: $message")
+          }
+        }
+
+        override def postStop(): Unit = {
+          unregisterEndpoint(endpoint.self)
+          safelyCall(endpoint) {
+            endpoint.onStop()
+          }
+        }
+
+        }), name = name)
+      endpointRef = new AkkaRpcEndpointRef(defaultAddress, actorRef, conf)
```

Comment:
@CodingCat Thanks. The lazy idea works. However, I still cannot use Akka's ExecutionContext because if AkkaRpcEndpointRef is sent to another node, it cannot get the `ExecutionContext` after deserialization.